src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs

```diff
@@ -22,12 +24,13 @@ internal ParsedExpression(IntermediateNode irnode, ScopeSymbol topScope, Culture
             _cultureInfo = cultureInfo ?? CultureInfo.CurrentCulture;
         }
 
-        public FormulaValue Eval(RecordValue parameters)
+        public Task<FormulaValue> EvalAsync(RecordValue parameters, CancellationToken cancel)
         {
-            var ev2 = new EvalVisitor(_cultureInfo);
+            var ev2 = new EvalVisitor(_cultureInfo, cancel);            
+
             var newValue = _irnode.Accept(ev2, SymbolContext.NewTopScope(_topScopeSymbol, parameters));
 
-            return newValue;
+            return newValue.AsTask();

```

Comment:
>.AsTask() [](http://example.com/codeflow?start=27&length=9)

ValueTask<> is has restrictions, so ok internally. Use for when the task is most commonly synchronous - to avoid heap allocations. 
 But public surface should use Task, so we call AsTask to convert.

Reply:
^^ should be in the summary of the ValueTask<> class?

Reply:
Please please put this into a comment. Always think of the "next guy" wondering what's going on, save their brain cycles :)